File: java/src/main/java/codingdojo/SafeCalculator.java
package codingdojo;

public class SafeCalculator {
  private final Authorizer authorizer;

  SafeCalculator(Authorizer authorizer) {
    this.authorizer = authorizer;
  }

  public int add(int a, int b) {
    var authorized = authorizer.authorize();
    if (authorized) { // <- bug here, should be if (!authorized)
      throw new UnauthorizedAccessException("Not authorized");
    }
    return a + b;
  }
}

File: go/safe_calculator.go
package calculator

type Calculator struct {
	authorizer IAuthorizer
}

func (c *Calculator) Add(x, y int) int {
	authorized := c.authorizer.Authorize()

	// BUG! Should be if !authorized
	if authorized {
		panic("Not authorized")
	}
	return x + y
}
